src/api/firebase.js

```diff
@@ -287,3 +291,25 @@ export async function getListOwnerDetails(userId, listName) {
 		throw error;
 	}
 }
+
+export async function deleteSingleList(userEmail, listPath) {
+	const userRef = doc(db, 'users', userEmail);
+	const listRef = doc(db, listPath);
+
+	const listDoc = await getDoc(listRef);
+	const userDoc = await getDoc(userRef);
+
+	if (listDoc.data().ownerID !== userDoc.data().uid) {
+		throw new Error('You are not the owner of this list.');
```

Comment:
Great job! If you wanted to be even more explicit, you could add "You can't delete this list because you are not the owner." 

It also brings up an interesting use-case. Perhaps in the future, you could give the user the option to hide a list if it's not theirs. It's not something I would tackle this week, as you are all busy meeting deadlines, but something interesting to consider if you choose to work on this once the cohort is done.

Reply:
Thanks, @polly89! I will make the text more explicit and consider this use case!